crates/transaction-generator-lib/src/workflow_delegator.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    account_generator::AccountGeneratorCreator, accounts_pool_wrapper::AccountsPoolWrapperCreator,
+    call_custom_modules::CustomModulesDelegationGeneratorCreator,
+    entry_points::EntryPointTransactionGenerator, EntryPoints, ObjectPool,
+    ReliableTransactionSubmitter, TransactionGenerator, TransactionGeneratorCreator, WorkflowKind,
+};
+use aptos_logger::{info, sample, sample::SampleRate};
+use aptos_sdk::{
+    transaction_builder::TransactionFactory,
+    types::{transaction::SignedTransaction, LocalAccount},
+};
+use std::{
+    cmp,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+#[derive(Clone)]
+enum StageTracking {
+    // stage is externally modified
+    ExternallySet(Arc<AtomicUsize>),
+    // we move to a next stage when all accounts have finished with the current stage
+    WhenDone(Arc<AtomicUsize>),
```

Comment:
nit: Is "InternallyManaged" a better name ?

Reply:
I think "WhenDone" explains more what it does. i.e. we could have other internally managed options  - like "TimerBased" etc.

Reply:
Ok